### PR TITLE
Error handling improvement

### PR DESCRIPTION
**Pull request by gchauvet**
This patch improve error statement handling, especially in JSONStream project.

### CODE REVIEW

**Comment by dominictarr** [comment]
@gchauvet you should be a bit more descriptive. saying it "improves error handling" is not as helpful as describing the specific way that it improves it. A very busy person will be reviewing this change, so you should make it as easy for them to say "merge" as possible. (quickly describe the problem this fixes, and reference any relavant issues incase the maintainer wants to see the original context - this will mean your PR can be merged much faster!)

**Comment by gchauvet** [comment]
The current jsonparse error handler implementation is incomplete. E.g, if a end-user override the "onError" method without calling the parent implementation, he will breaks the current behavior (see dominictarr/JSONStream#66).
This case introduce an integrity break: next data performed by the state machine after the catching of parser exception are not guaranteed.

I propose a new error recovery mechanism to the state machine by implementing the panic mode with the introduction of a new terminal state called 'STOP'.
The advantages are to keep the current error behavior and improving the internal state machine workflow.

From end-user perspective, that's means the parser must be reinstancied for a new processing (It would possible to add a method to reset the state machine).

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "jsonparse",
   "description": "This is a pure-js JSON streaming parser for node.js",
   "tags": ["json", "stream"],
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (creationix) [inline package.json]: Since this is a behavior change with new error handling semantics, we need to at lease bump the middle number.

**Comment by creationix** [comment]
I just got back from vacation and and swamped catching up on stuff.  If you're sure this doesn't break anything and are willing to be responsible for any problems people may come across, I can merge it for you.

**Comment by gchauvet** [comment]
In my point of view, this patch is a good enhancement. I've incremented the project version to 1.1.0.
But if a user really needs the older version, he just need to force jsonparse to 1.0.x version.

**Comment by creationix** [comment]
Published to npm as jsonparse@1.1.0.

**Pull request merged**